### PR TITLE
Zebra-puzzle: add test template

### LINKED ISSUE #2004: zebra-puzzle: add test template
## Task
Create a test template file for `zebra-puzzle`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `zebra-puzzle-add-test-template`
    - Run: `git checkout -b zebra-puzzle-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/zebra-puzzle/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py zebra-puzzle`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by crsmi**
This pull request adds test template for the zebra-puzzle exercise.
Resolves #2004

### CODE REVIEW

File: exercises/zebra-puzzle/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{%- import "generator_macros.j2" as macros with context -%}
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        expected = {{ case["expected"] }}
+        self.assertEqual({{ case["property"] }}(), expected)
```
Comment (cmccandless) [inline exercises/zebra-puzzle/.meta/template.j2]: How about:
```
self.assertEqual({{ case["property"] }}(), {{ case["expected"] }})
```

Comment (crsmi) [reply exercises/zebra-puzzle/.meta/template.j2]: I changed before I saw this.

```
expected = "{{ case["expected"] }}"
'''

is this okay?

**Comment by crsmi** [comment]
It looks like the "property" values for the cases in the `canonical-data.json` file differ from the function names in `zebra_puzzle.py`.

I'm guessing this requires a fix in [problem-specifications](../../problem-specifications), I'll look at making that --  if that is okay.

File: exercises/zebra-puzzle/.meta/template.j2 (hunk 1 above)
Comment (cmccandless) [reply exercises/zebra-puzzle/.meta/template.j2]: I think it'd still be nice to get this on one line in the generated output, like in the original. I'm mass-reviewing PRs for test templates, I'm much less likely to miss something if the diff is smaller.

File: exercises/zebra-puzzle/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{%- import "generator_macros.j2" as macros with context -%}
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        expected = "{{ case["expected"] }}"
+        self.assertEqual({{ case["property"] }}(), expected)
```
Comment (cmccandless) [inline exercises/zebra-puzzle/.meta/template.j2]: You can fix the casing issue between the solution and the spec with the following syntax:
 ```
        self.assertEqual({{ case["property"] | to_snake }}(), expected)
```

Comment (crsmi) [reply exercises/zebra-puzzle/.meta/template.j2]: That is super convenient!

File: exercises/zebra-puzzle/.meta/template.j2 (hunk 1 above)
Comment (crsmi) [reply exercises/zebra-puzzle/.meta/template.j2]: That makes sense. I think everything is good here now. Thanks for the help!

**Pull request merged**